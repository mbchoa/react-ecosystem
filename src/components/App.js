import React, { Component } from 'react';
import PropTypes from 'prop-types';

import HelloWorld from './HelloWorld';

class App extends Component {
    static childContextTypes = {
        insertCss: PropTypes.func.isRequired
    };

    getChildContext() {
        return {
            insertCss: (...styles) => {
                const removeCssArr = styles.map(x => x._insertCss());
                return () => { removeCssArr.forEach(removeCss => removeCss()); }
            }
        };
    }

    render () {
        return (
            <div>
                <HelloWorld />
            </div>
        );
    }
}

export default App;
