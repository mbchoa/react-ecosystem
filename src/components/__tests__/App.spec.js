import React from 'react';
import { shallow } from 'enzyme';

import App from '../App';

describe('App', () => {
    function renderComponent () {
        return shallow(<App />);
    }

    it('should render div with content of "Hello from the React Ecosystem!"', () => {
        expect(renderComponent()).toMatchSnapshot();
    });
});
