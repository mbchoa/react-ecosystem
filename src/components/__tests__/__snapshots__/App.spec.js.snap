// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render div with content of "Hello from the React Ecosystem!" 1`] = `
<div>
  Hello from the React Ecosystem!
</div>
`;
